lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControlConfig.java

```diff
@@ -14,36 +14,48 @@
 package io.trino.plugin.base.security;
 
 import io.airlift.configuration.Config;
-import io.airlift.configuration.validation.FileExists;
+import io.airlift.configuration.ConfigDescription;
 import io.airlift.units.Duration;
 import io.airlift.units.MinDuration;
 
+import javax.validation.constraints.AssertTrue;
 import javax.validation.constraints.NotNull;
 
 import java.io.File;
+import java.util.Objects;
 
 public class FileBasedAccessControlConfig
 {
     public static final String SECURITY_CONFIG_FILE = "security.config-file";
     public static final String SECURITY_REFRESH_PERIOD = "security.refresh-period";
+    public static final String SECURITY_JSON_POINTER = "security.json-pointer";
 
-    private File configFile;
+    private String configFile;
     private Duration refreshPeriod;
+    private String jsonPointer = "";
 
     @NotNull
-    @FileExists
-    public File getConfigFile()
+    public String getConfigFile()
     {
         return configFile;
     }
 
     @Config(SECURITY_CONFIG_FILE)
-    public FileBasedAccessControlConfig setConfigFile(File configFile)
+    public FileBasedAccessControlConfig setConfigFile(String configFile)
     {
         this.configFile = configFile;
         return this;
     }
 
+    @AssertTrue(message = "Config file does not exist.")
+    public boolean isConfigFileValid()
+    {
+        if (Objects.nonNull(configFile) && !isHttp() && !new File(configFile).exists()) {
```

Comment:
hmm not sure if we don't need non-null check here. We have NotNull validation for configFile field. You shouldn't relay on order of validations, If I remove this non-null check we might (and will in this case) end up with NPE instead of NotNull validation failure

Reply:
Custom validation happens after the getter/setter validation happens - so `NotNull` will be checked already. I think this non-null check would be redundant